### ISSUE TITLE
Updated vault on android and lost all accounts and unable to recover them!

### ISSUE THREAD

**Issue by shanness** [airgap-it/airgap-vault#96]
Hi,

I updated my wallet (to the app image from running it using npm).  This lost all the account data, but no biggie, I added again from vault.

Then I updated vault on Android, and it lost everything!  I did a "Import" with my seed phrase, but when I go to "Add Account" there are no coins listed at all (it's just blank between that and Advanced Mode).

I created a new private key and had exactly the same problem (unable to create accounts because there are no coins shown)
About says :-
34171
3.10.0

Also, the sync barcode doesn't work on the AppImage nor the npm serve version of the wallet on Linux.

Basically I've lost access to my crypto!

Galaxy S6 if that makes any difference

**Comment by shanness** [comment]
Stephen on telegram helped me.  Basically adding accounts is broken on version 3.10 (no coins show in Add Account), downgrading to version 3.9 fixed it.
![image](https://user-images.githubusercontent.com/631580/134926398-83e353cb-8376-4c04-ab6d-368695574917.png)

It's running Android 5.0.1 (which is up to date according to the phone - Galaxy S6)

**Comment by lukeisontheroad** [comment]
Hi Shanness, we are trying to support as many devices as possible, however, especially on the Android front this can be an exhausting task. A list of recommended devices can be found in our documentation: https://support.airgap.it/airgap-vault/recommended-devices

**Comment by shanness** [comment]
Hey @lukeisontheroad , yeah, I was sent that list on telegram, but it only has 3 phones on it, and the motorola is about $200 in australia,  the galaxy is $400+ and the pixel 5 is $1500 (!!).  Hence I'm looking for a replacement to my S6 which costs something in the same range as a ledger..

**Comment by shanness** [comment]
Seems the lack of documented supported old/cheap phones is probably severely limiting uptake of airgap to me..  Would be a good thing to crowdsource from your users?

**Comment by AndreasGassmann** [comment]
The list of the recommended devices are devices that we used and verified that it works, plus we tried to categorise them based on some features, eg. vendor or price, so there is something in there for everyone. But as you mention, prices vary widely around the globe, so it's impossible for us to make a list of recommended devices that works for everyone. The Moto E6 is $40 in the US and over $200 in Australia. Maybe there is one in Australia for $50, but that one would be over $200 in Europe, etc.

We have another list where we are trying to collect all the devices that support and don't support AirGap Vault. Here we are happy for any additions, but so far the community engagement hasn't been high and the list is still short: https://support.airgap.it/airgap-vault/supported-devices

**Comment by shanness** [comment]
I'm back to the same problem!  My seed disappeared again, re-entered it, but can't add coins.  Despite still being on 3.9 that Stephen helped me with that I mentioned above.  Now 3.9 is giving the same problem as the screenshot above.  I've tried updating webview, but to no avail..  And this seeming bug in the wallet drove me crazy for about 30 mins (I assume empty vaults get ignored by the wallet?)
https://github.com/airgap-it/airgap-wallet/issues/112

Anyhow, yet again, I've lost all access to my crypto...

**Comment by Originalimoc** [comment]
I do clean install every upgrade. So, not a issue to me😂

**Comment by AndreasGassmann** [comment]
@shanness Sorry for the late reply and sorry to hear you are having problems again.

So let me make a summary. Your device is the `Samsung S6, Android 5.0.1 (latest version), updated webview`. You have tried with `3.9.0`, `3.10.0` and presumably `3.12.0`.

- Regarding the webview, I don't remember if on 5.0.1 there is still the integrated webview or if it is already managed by the chrome app. Have you tried to install/update both? Follow this guide: https://support.airgap.it/guides/update-webview

- You initially had problems on `3.10.0`, but downgrading to `3.9.0` fixed it. Did you now update again, eg. to `3.12.0`? If not, did you do anything else with the device that would trigger the regression?

- On `3.12.0` we introduced a new "error history" view, which shows you the latest local errors for debugging. Is there anything interesting showing up on that list around the time when you try to start the apps or try to import secrets?

- Did you try to do a clean install (removing cache and app data, then installing `3.12.0` again)?

If you did not actually update the Vault application and you are experiencing this issue "randomly", then it's almost certainly related to your device. Especially older devices can have very odd and unexpected behaviour, primarily around the secure storage, which is sometimes just wiped randomly or inaccessible. This can have various reasons, eg. if no device PIN is set, but it can also simply be random.

We will now again run some tests on our older devices and try to reproduce this issue on our end.
